sections/futures/MobileTrade/drawers/TradeConfirmationDrawer.tsx

```diff
@@ -0,0 +1,149 @@
+	const market = useRecoilValue(currentMarketState);
+	const { useExchangeRatesQuery, useEthGasPriceQuery } = useSynthetixQueries();
+	const { selectedPriceCurrency } = useSelectedPriceCurrency();
+	const ethGasPriceQuery = useEthGasPriceQuery();
+	const exchangeRatesQuery = useExchangeRatesQuery();
+	const { data: potentialTradeDetails } = useGetFuturesPotentialTradeDetails();
+
+	const exchangeRates = useMemo(
+		() => (exchangeRatesQuery.isSuccess ? exchangeRatesQuery.data ?? null : null),
+		[exchangeRatesQuery.isSuccess, exchangeRatesQuery.data]
+	);
+
+	const ethPriceRate = useMemo(
+		() => newGetExchangeRatesForCurrencies(exchangeRates, Synths.sETH, selectedPriceCurrency.name),
+		[exchangeRates, selectedPriceCurrency.name]
+	);
+
+	const gasPrice = ethGasPriceQuery.data != null ? ethGasPriceQuery.data[gasSpeed] : null;
+
+	const transactionFee = useMemo(
+		() => newGetTransactionPrice(gasPrice, gasLimit, ethPriceRate, l1Fee),
+		[gasPrice, gasLimit, ethPriceRate, l1Fee]
+	);
+
+	const positionDetails = useMemo(() => {
+		return potentialTradeDetails
+			? {
+					...potentialTradeDetails,
+					size: potentialTradeDetails.size.abs(),
+					side: potentialTradeDetails.size.gte(zeroBN) ? PositionSide.LONG : PositionSide.SHORT,
+					leverage: potentialTradeDetails.margin.eq(zeroBN)
+						? zeroBN
+						: potentialTradeDetails.size
+								.mul(potentialTradeDetails.price)
+								.div(potentialTradeDetails.margin)
+								.abs(),
+			  }
+			: null;
+	}, [potentialTradeDetails]);
+
+	const dataRows = useMemo(
+		() => [
+			{ label: 'side', value: (positionDetails?.side ?? PositionSide.LONG).toUpperCase() },
+			{
+				label: 'size',
+				value: formatCurrency(market || '', positionDetails?.size ?? zeroBN, {
+					sign: market ? synthsMap[market]?.sign : '',
+				}),
+			},
+			{ label: 'leverage', value: `${formatNumber(positionDetails?.leverage ?? zeroBN)}x` },
+			{
+				label: 'current price',
+				value: formatCurrency(Synths.sUSD, positionDetails?.price ?? zeroBN, { sign: '$' }),
+			},
+			{
+				label: 'liquidation price',
+				value: formatCurrency(Synths.sUSD, positionDetails?.liqPrice ?? zeroBN, {
+					sign: '$',
+				}),
+			},
+			{
+				label: 'margin',
+				value: formatCurrency(Synths.sUSD, positionDetails?.margin ?? zeroBN, { sign: '$' }),
+			},
+			{
+				label: 'protocol fee',
+				value: formatCurrency(Synths.sUSD, positionDetails?.fee ?? zeroBN, { sign: '$' }),
+			},
+			{
+				label: 'network gas fee',
+				value: formatCurrency(selectedPriceCurrency.name as CurrencyKey, transactionFee ?? zeroBN, {
+					sign: '$',
+					minDecimals: 2,
+				}),
+			},
+		],
+		[positionDetails, market, synthsMap, transactionFee, selectedPriceCurrency]
+	);
+
+	return (
+		<BaseDrawer
+			open={open}
+			closeDrawer={closeDrawer}
+			items={dataRows}
+			buttons={
+				<ConfirmTradeButton
+					variant="primary"
+					isRounded
+					onClick={() => {
+						onConfirmOrder();
+						closeDrawer();
+					}}
```

Comment:
Would be cool if this reset the inputs as well. There is no indication that there is a pending transaction, until it goes through and the position updates.

Reply:
Another option would be to leave the drawer open until the transaction is complete. While it's pending, place a loading spinner in the button. When it succeeds, close the drawer.

Reply:
I think that makes sense. @asherism maybe we should add loading states for the buttons?